framework/src/bcs/NodalBC.C

```diff
@@ -43,6 +77,13 @@ NodalBC::computeResidual(NumericVector<Number> & residual)
     dof_id_type & dof_idx = _var.nodalDofIndex();
     _qp = 0;
     residual.set(dof_idx, computeQpResidual());
+
+    if (_has_save_in)
+    {
+      Threads::spin_mutex::scoped_lock lock(Threads::spin_mtx);
+      for (unsigned int i=0; i<_save_in.size(); i++)
+        _save_in[i]->sys().solution().set(dof_idx, 0);
```

Comment:
Can I add a function like `FEProblem::hasNodalSaveIn()`, which will return a flag set during adding `NodalBC` with `save_in=true` or `diag_save_in=true`? Then we can call `close()` if the function returns true.